File: sdk/src/main/java/org/p2p/solanaj/rpc/types/TokenAccountBalance.java
package org.p2p.solanaj.rpc.types;

import java.math.BigDecimal;
import java.math.BigInteger;

import com.squareup.moshi.Json;

import org.p2p.solanaj.rpc.types.RpcResultObject;

public class TokenAccountBalance extends RpcResultObject {

    public static class Balance {
        @Json(name = "amount")
        private String amount;
        @Json(name = "decimals")
        private int decimals;

        public String getAmount() {
            return amount;
        }

    }

    @Json(name = "value")
    private Balance value;

    public String getAmountString() {
        return value.getAmount();
    }

    public BigInteger getAmount() {
        return new BigInteger(value.getAmount());
    }

}

File: solanaj/src/main/java/org/p2p/solanaj/model/types/TokenAccountBalance.kt
package org.p2p.solanaj.model.types

import com.google.gson.annotations.SerializedName
import java.math.BigInteger

data class TokenAccountBalance(
    @SerializedName("value")
    val value: Balance
) : RpcResultObject() {

    data class Balance(
        @SerializedName("amount")
        val amount: String,

        @SerializedName("decimals")
        val decimals: Int
    )

    val amount: BigInteger
        get() = BigInteger(value.amount)
}
